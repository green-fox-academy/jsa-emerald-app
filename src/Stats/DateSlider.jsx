import React from 'react';
import { View, Text } from 'react-native';
import PropTypes from 'prop-types';
import { Button } from 'react-native-elements';

export default function DateSlider(props) {
  const { viewSet, onPressBtn, viewType } = props;
  const format = viewType === 'month' ? 'MMM YYYY' : 'YYYY';

  return (
    <View style={{ flexDirection: 'row', justifyContent: 'space-around' }}>
      <Button
        containerStyle={{ flex: 1, borderBottomWidth: 2, borderBottomColor: 'white' }}
        title={viewSet[0].format(format)}
        type="clear"
        titleStyle={{ color: 'grey' }}
        onPress={() => onPressBtn(viewSet[0], viewType)}
      />
      <View style={{
        flex: 1, borderBottomWidth: 2, borderBottomColor: '#30d29d', justifyContent: 'center',
      }}
      >
        <Text style={{ textAlign: 'center', fontSize: 18, fontWeight: '500' }}>{viewSet[1].format(format)}</Text>
      </View>
      <Button
        containerStyle={{ flex: 1, borderBottomWidth: 2, borderBottomColor: 'white' }}
        title={viewSet[2].format(format)}
        type="clear"
        titleStyle={{ color: 'grey' }}
        onPress={() => onPressBtn(viewSet[2], viewType)}
      />
    </View>
  );
}

DateSlider.propTypes = {
  viewSet: PropTypes.instanceOf(Array),
  onPressBtn: PropTypes.func.isRequired,
  viewType: PropTypes.oneOf(['month', 'year']),
};

DateSlider.defaultProps = {
  viewSet: [],
  viewType: 'month',
};
