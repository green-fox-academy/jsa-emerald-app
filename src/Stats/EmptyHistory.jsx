import React from 'react';
import { View } from 'react-native';
import { Image } from 'react-native-elements';
import { useSelector } from 'react-redux';
import setThemeStyle from '../Common/theme/setThemeStyle';

const img = require('../../assets/empty_box.png');

export default function EmptyHistory() {
  const { themeMode } = useSelector((state) => state.theme);
  const styles = setThemeStyle(themeMode);

  return (
    <View style={[styles.card, { alignItems: 'center', justifyContent: 'center', paddingVertical: 30 }]}>
      <Image
        source={img}
        style={{ width: 220, height: 220 }}
      />
    </View>
  );
}
