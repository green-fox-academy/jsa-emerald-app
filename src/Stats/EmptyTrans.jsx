import React from 'react';
import { View, Text } from 'react-native';

export default function EmptyTrans() {
  return (<View><Text>You haven&quot;t added any </Text></View>);
}
