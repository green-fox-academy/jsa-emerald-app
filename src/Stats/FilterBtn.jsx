import React from 'react';
import LargeButton from '../Common/LargeBtn';

export default function FilterBtn() {
  return (
    <>
      <LargeButton subtitle="Spent" title="-$400.00" note="Expense" bgColor="red" numberOfBtns={2} />
      <LargeButton subtitle="Earned" title="+$680.00" note="Incomes" bgColor="green" numberOfBtns={2} />
    </>
  );
}
