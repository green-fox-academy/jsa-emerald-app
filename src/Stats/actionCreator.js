export const actionType = {
  ADD_TRANS: 'ADD_TRANS',
};

export function addNewTransaction(type, date, amount) {
  return {
    type: actionType.ADD_TRANS,
    data: {
      type, date, amount,
    },
  };
}
