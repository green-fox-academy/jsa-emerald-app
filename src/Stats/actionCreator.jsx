export const actionType = {
  ADD_TRANS: 'ADD_TRANS',
  SET_NEW_TRANS_TYPE: 'SET_NEW_TRANS_TYPE',
  SET_NEW_TRANS_DATE: 'SET_NEW_TRANS_DATE',
  SET_NEW_TRANS_AMOUNT: 'SET_NEW_TRANS_AMOUNT',
  SET_NEW_TRANS_INSERTION_SUCCESS: 'SET_NEW_TRANS_INSERTION_SUCCESS',
};

export function addNewTransaction(type, date, amount) {
  return {
    type: actionType.ADD_TRANS,
    data: {
      type, date, amount,
    },
  };
}

export function setNewTransactionType(type) {
  return {
    type: actionType.SET_NEW_TRANS_TYPE,
    data: type,
  };
}

export function setNewTransactionAmount(amount) {
  return {
    type: actionType.SET_NEW_TRANS_AMOUNT,
    data: amount,
  };
}

export function setNewTransactionDate(date) {
  return {
    type: actionType.SET_NEW_TRANS_DATE,
    data: date,
  };
}

export function setNewTransactionInsertionSuccess(bool) {
  return {
    type: actionType.SET_NEW_TRANS_INSERTION_SUCCESS,
    data: bool,
  };
}
