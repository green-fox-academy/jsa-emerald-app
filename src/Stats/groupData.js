const moment = require('moment');

const groupData = (transactions) => {
  const sortedData = transactions.sort((a, b) => (b.date - a.date));
  const convertedSet = sortedData.map((value) => ({ amount: value.amount, date: moment.unix(value.date).format('MMMM Do YYYY'), type: value.type }));
  const result = []; let temp = [convertedSet[0]];
  for (let i = 1; i < convertedSet.length; i += 1) {
    if (temp[0].date === convertedSet[i].date) {
      temp.push(convertedSet[i]);
    } else {
      result.push(temp);
      temp = [convertedSet[i]];
    }
  }
  if (temp.length > 0) {
    result.push(temp);
  }
  return result;
};

export default groupData;
