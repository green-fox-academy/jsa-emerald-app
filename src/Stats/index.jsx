import React, { useState } from 'react';
import { View, ScrollView } from 'react-native';
import { ListItem, Overlay, Button } from 'react-native-elements';
import { useSelector } from 'react-redux';
import styles from '../Common/themeStyle';
import MainHeader from '../Common/MainHeader';
import utils from './utils';
import DateSlider from './DateSlider';
import FilterBtn from './FilterBtn';
import TransList from './TransList';
import EmptyHistory from './EmptyHistory';
import ViewSelectionItem from './ViewSelectionItem';

const moment = require('moment');

export default function Stats() {
  const { transactions } = useSelector((state) => state.transactions);
  const [view, setCurrentView] = useState('month');
  const [timePeriodOptions, setTimePeriod] = useState(utils.getDateSet(moment(), view));
  const [isOverlayVisible, setOverlayVisibility] = useState(false);
  const [transFilter, setTransFilter] = useState('all');

  const updateHeaderView = (type) => {
    setCurrentView(type);
    setTimePeriod(utils.getDateSet(moment(), type));
    setOverlayVisibility(!isOverlayVisible);
  };

  const filterTransactions = (transactionsList) => {
    const transactionsWithSpecificType = utils.filterTransactionByType(
      transactionsList,
      transFilter,
    );
    return utils.filterTransactionsByDate(
      transactionsWithSpecificType,
      timePeriodOptions[1],
      view,
    );
  };

  const calculateSumByType = (dataList, type) => dataList
    .filter((item) => item.type === type)
    .reduce((sum, { amount }) => sum + parseFloat(amount), 0);

  const filteredTransactions = filterTransactions(transactions);
  const processedTransactions = utils.groupTransactionsByDate(filteredTransactions);
  const totalExpense = calculateSumByType(filteredTransactions, 'Expense');
  const totalIncome = calculateSumByType(filteredTransactions, 'Income');

  return (
    <View style={{ flex: 1, flexDirection: 'column' }}>
      <Overlay height={200} isVisible={isOverlayVisible}>
        <View>
          <View style={{ flexDirection: 'row', justifyContent: 'flex-end' }}>
            <Button
              id="btn-stats-index-vsb"
              icon={{ name: 'close' }}
              type="clear"
              onPress={() => setOverlayVisibility(false)}
            />
          </View>
          <View style={{ marginTop: 20 }}>
            <ViewSelectionItem title="Month" pressHandler={updateHeaderView} />
            <ViewSelectionItem title="Year" pressHandler={updateHeaderView} />
          </View>
        </View>
      </Overlay>
      <MainHeader title="Activity" onPressBtn={() => setOverlayVisibility(true)} btnName="filter" />
      <DateSlider
        viewSet={timePeriodOptions}
        onPressBtn={(value, type) => setTimePeriod(utils.getDateSet(value, type))}
        viewType={view}
      />
      <ScrollView style={styles.deviceBody}>
        <View style={{ marginBottom: 20 }}>
          <View style={{ flexDirection: 'row', justifyContent: 'space-between' }}>
            <FilterBtn
              currentFilter={transFilter}
              onFilterChange={(newFilter) => (newFilter === transFilter ? setTransFilter('all') : setTransFilter(newFilter))}
              totalExpense={totalExpense}
              totalIncome={totalIncome}
            />
          </View>
          {(transactions.length !== 0)
            ? (
              <TransList
                transactions={processedTransactions}
              />
            )
            : <EmptyHistory />}
        </View>
      </ScrollView>
    </View>
  );
}
