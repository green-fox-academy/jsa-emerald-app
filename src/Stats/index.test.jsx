import React from 'react';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import Stats from './index';

const mockStore = configureStore([]);
let store;
beforeAll(() => {
  store = mockStore({
    transactions: { transactions: [] },
  });
});

describe('Stats', () => {
  it('render', () => {
    const wrapper = mount(
      <Provider store={store}>
        <Stats />
      </Provider>,
    );
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
