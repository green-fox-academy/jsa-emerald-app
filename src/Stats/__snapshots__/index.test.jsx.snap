// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats render 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <Modal
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    theme={
      Object {
        "colors": Object {
          "disabled": "hsl(208, 8%, 90%)",
          "divider": "#bcbbc1",
          "error": "#ff190c",
          "grey0": "#393e42",
          "grey1": "#43484d",
          "grey2": "#5e6977",
          "grey3": "#86939e",
          "grey4": "#bdc6cf",
          "grey5": "#e1e8ee",
          "greyOutline": "#bbb",
          "platform": Object {
            "android": Object {
              "error": "#f44336",
              "primary": "#2196f3",
              "secondary": "#9C27B0",
              "success": "#4caf50",
              "warning": "#ffeb3b",
            },
            "ios": Object {
              "error": "#ff3b30",
              "primary": "#007aff",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
          },
          "primary": "#2089dc",
          "searchBg": "#303337",
          "secondary": "#8F0CE8",
          "success": "#52c41a",
          "warning": "#faad14",
        },
      }
    }
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, .4)",
          "height": "100%",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        }
      }
      testID="RNE__Overlay__backdrop"
    />
    <View
      pointerEvents="box-none"
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 3,
            "height": 200,
            "padding": 10,
            "shadowColor": "rgba(0, 0, 0, .3)",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowRadius": 4,
            "width": 670,
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "flex-end",
              }
            }
          >
            <View
              style={
                Object {
                  "borderRadius": 3,
                }
              }
            >
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "borderColor": "#2089dc",
                      "borderRadius": 3,
                      "borderWidth": 0,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "padding": 8,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "marginHorizontal": 5,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "justifyContent": "center",
                        }
                      }
                      theme={
                        Object {
                          "colors": Object {
                            "disabled": "hsl(208, 8%, 90%)",
                            "divider": "#bcbbc1",
                            "error": "#ff190c",
                            "grey0": "#393e42",
                            "grey1": "#43484d",
                            "grey2": "#5e6977",
                            "grey3": "#86939e",
                            "grey4": "#bdc6cf",
                            "grey5": "#e1e8ee",
                            "greyOutline": "#bbb",
                            "platform": Object {
                              "android": Object {
                                "error": "#f44336",
                                "primary": "#2196f3",
                                "secondary": "#9C27B0",
                                "success": "#4caf50",
                                "warning": "#ffeb3b",
                              },
                              "ios": Object {
                                "error": "#ff3b30",
                                "primary": "#007aff",
                                "secondary": "#5856d6",
                                "success": "#4cd964",
                                "warning": "#ffcc00",
                              },
                            },
                            "primary": "#2089dc",
                            "searchBg": "#303337",
                            "secondary": "#8F0CE8",
                            "success": "#52c41a",
                            "warning": "#faad14",
                          },
                        }
                      }
                      underlayColor="white"
                    >
                      <Text />
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "marginTop": 20,
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={null}
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderBottomWidth": 0.5,
                    "borderColor": "#bcbbc1",
                    "borderTopWidth": 0.5,
                    "flexDirection": "row",
                    "padding": 14,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "fontSize": 17,
                      }
                    }
                    testID="listItemTitle"
                    theme={
                      Object {
                        "colors": Object {
                          "disabled": "hsl(208, 8%, 90%)",
                          "divider": "#bcbbc1",
                          "error": "#ff190c",
                          "grey0": "#393e42",
                          "grey1": "#43484d",
                          "grey2": "#5e6977",
                          "grey3": "#86939e",
                          "grey4": "#bdc6cf",
                          "grey5": "#e1e8ee",
                          "greyOutline": "#bbb",
                          "platform": Object {
                            "android": Object {
                              "error": "#f44336",
                              "primary": "#2196f3",
                              "secondary": "#9C27B0",
                              "success": "#4caf50",
                              "warning": "#ffeb3b",
                            },
                            "ios": Object {
                              "error": "#ff3b30",
                              "primary": "#007aff",
                              "secondary": "#5856d6",
                              "success": "#4cd964",
                              "warning": "#ffcc00",
                            },
                          },
                          "primary": "#2089dc",
                          "searchBg": "#303337",
                          "secondary": "#8F0CE8",
                          "success": "#52c41a",
                          "warning": "#faad14",
                        },
                      }
                    }
                  >
                    Month
                  </Text>
                </View>
              </View>
            </View>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={null}
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderBottomWidth": 0.5,
                    "borderColor": "#bcbbc1",
                    "flexDirection": "row",
                    "padding": 14,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "fontSize": 17,
                      }
                    }
                    testID="listItemTitle"
                    theme={
                      Object {
                        "colors": Object {
                          "disabled": "hsl(208, 8%, 90%)",
                          "divider": "#bcbbc1",
                          "error": "#ff190c",
                          "grey0": "#393e42",
                          "grey1": "#43484d",
                          "grey2": "#5e6977",
                          "grey3": "#86939e",
                          "grey4": "#bdc6cf",
                          "grey5": "#e1e8ee",
                          "greyOutline": "#bbb",
                          "platform": Object {
                            "android": Object {
                              "error": "#f44336",
                              "primary": "#2196f3",
                              "secondary": "#9C27B0",
                              "success": "#4caf50",
                              "warning": "#ffeb3b",
                            },
                            "ios": Object {
                              "error": "#ff3b30",
                              "primary": "#007aff",
                              "secondary": "#5856d6",
                              "success": "#4cd964",
                              "warning": "#ffcc00",
                            },
                          },
                          "primary": "#2089dc",
                          "searchBg": "#303337",
                          "secondary": "#8F0CE8",
                          "success": "#52c41a",
                          "warning": "#faad14",
                        },
                      }
                    }
                  >
                    Year
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "flexDirection": "row",
        "height": 100,
        "justifyContent": "space-between",
        "paddingBottom": 20,
        "paddingLeft": 20,
        "paddingRight": 20,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "borderRadius": 3,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderColor": "#2089dc",
                "borderRadius": 3,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 8,
              }
            }
          />
        </View>
      </View>
    </View>
    <Text
      style={
        Array [
          Object {
            "fontSize": 20,
            "fontWeight": "500",
            "textAlign": "center",
          },
          Object {
            "flex": 1,
          },
        ]
      }
    >
      Activity
    </Text>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "borderRadius": 3,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderColor": "#2089dc",
                "borderRadius": 3,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 8,
              }
            }
          >
            <View
              style={
                Object {
                  "marginHorizontal": 5,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
                theme={
                  Object {
                    "colors": Object {
                      "disabled": "hsl(208, 8%, 90%)",
                      "divider": "#bcbbc1",
                      "error": "#ff190c",
                      "grey0": "#393e42",
                      "grey1": "#43484d",
                      "grey2": "#5e6977",
                      "grey3": "#86939e",
                      "grey4": "#bdc6cf",
                      "grey5": "#e1e8ee",
                      "greyOutline": "#bbb",
                      "platform": Object {
                        "android": Object {
                          "error": "#f44336",
                          "primary": "#2196f3",
                          "secondary": "#9C27B0",
                          "success": "#4caf50",
                          "warning": "#ffeb3b",
                        },
                        "ios": Object {
                          "error": "#ff3b30",
                          "primary": "#007aff",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                      },
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#8F0CE8",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  }
                }
                underlayColor="white"
              >
                <Text />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-around",
      }
    }
  >
    <View
      style={
        Object {
          "borderBottomColor": "white",
          "borderBottomWidth": 2,
          "borderRadius": 3,
          "flex": 1,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "grey",
                "fontSize": 18,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Nov 2019
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "#30d29d",
          "borderBottomWidth": 2,
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 18,
            "fontWeight": "500",
            "textAlign": "center",
          }
        }
      >
        Dec 2019
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "white",
          "borderBottomWidth": 2,
          "borderRadius": 3,
          "flex": 1,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "grey",
                "fontSize": 18,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Jan 2020
          </Text>
        </View>
      </View>
    </View>
  </View>
  <RCTScrollView
    style={
      Object {
        "backgroundColor": "#f3f5fa",
        "flexDirection": "column",
        "paddingLeft": 20,
        "paddingRight": 20,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "marginBottom": 20,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "elevation": 8,
                  "marginTop": 20,
                  "shadowColor": "#f38aa9",
                  "shadowOffset": Object {
                    "height": 6,
                    "width": 0,
                  },
                  "shadowOpacity": 0.7,
                  "shadowRadius": 4.65,
                },
                Object {
                  "width": "48%",
                },
              ]
            }
          >
            <ViewManagerAdapter_ExpoLinearGradient
              proxiedProperties={
                Object {
                  "colors": Array [
                    4293488030,
                    4294945423,
                  ],
                  "endPoint": Array [
                    0.9,
                    0.1,
                  ],
                  "locations": undefined,
                  "startPoint": Array [
                    0.1,
                    0.9,
                  ],
                }
              }
              style={
                Object {
                  "borderRadius": 10,
                  "padding": 20,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 15,
                    "fontWeight": "300",
                  }
                }
              >
                Spent
              </Text>
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 18,
                    "fontWeight": "400",
                    "marginBottom": 5,
                    "marginTop": 5,
                  }
                }
              >
                -$400.00
              </Text>
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 15,
                    "fontWeight": "300",
                  }
                }
              >
                Expense
              </Text>
            </ViewManagerAdapter_ExpoLinearGradient>
          </View>
          <View
            style={
              Array [
                Object {
                  "elevation": 8,
                  "marginTop": 20,
                  "shadowColor": "#70ea93",
                  "shadowOffset": Object {
                    "height": 6,
                    "width": 0,
                  },
                  "shadowOpacity": 0.7,
                  "shadowRadius": 4.65,
                },
                Object {
                  "width": "48%",
                },
              ]
            }
          >
            <ViewManagerAdapter_ExpoLinearGradient
              proxiedProperties={
                Object {
                  "colors": Array [
                    4281321625,
                    4286115731,
                  ],
                  "endPoint": Array [
                    0.9,
                    0.1,
                  ],
                  "locations": undefined,
                  "startPoint": Array [
                    0.1,
                    0.9,
                  ],
                }
              }
              style={
                Object {
                  "borderRadius": 10,
                  "padding": 20,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 15,
                    "fontWeight": "300",
                  }
                }
              >
                Earned
              </Text>
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 18,
                    "fontWeight": "400",
                    "marginBottom": 5,
                    "marginTop": 5,
                  }
                }
              >
                +$680.00
              </Text>
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 15,
                    "fontWeight": "300",
                  }
                }
              >
                Incomes
              </Text>
            </ViewManagerAdapter_ExpoLinearGradient>
          </View>
        </View>
        <View>
          <Text>
            You haven"t added any transactions yet
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
